### PR TITLE
Throttle FunctionController calls

### PR DESCRIPTION
**Pull request by davidebbo**
Not as fancy as doing caching, but simpler.

### CODE REVIEW

**Comment by davidebbo** [comment]
Note that we want want to solve this at a different level like geo. Only having this here for reference in case we go down that route.

File: Kudu.Services/Filters/RequestThrottleActionFilterAttribute.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+using System.Web.Http.Controllers;
+using System.Web.Http.Filters;
+
+namespace Kudu.Services.Filters
+{
+    public sealed class RequestThrottleActionFilterAttribute : ActionFilterAttribute
+    {
+        int _concurrency = 0;
+
+        public int MaxConcurrency { get; set; } = 5;
```
Comment (suwatch) [inline Kudu.Services/Filters/RequestThrottleActionFilterAttribute.cs]: Wonder if this should be appSettings with upper limit at, say, 20 pending requests.  I understand bad usages can abuse it (they can anyway by deploying private kudu).  On the flip side, it gives us a quick knob to mitigate if regression.     We can ship with appSettings and then look at log if normal usage will ever get to this and make it const.

File: Kudu.Services/Filters/RequestThrottleActionFilterAttribute.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+        public override void OnActionExecuting(HttpActionContext actionContext)
+        {
+            // If we've already reached the concurrency limit, return 429 (too many requests)
+            // and ask to retry after 5 seconds
+            if (_concurrency >= MaxConcurrency)
+            {
+                actionContext.Response = new HttpResponseMessage((HttpStatusCode)429);
```
Comment (suwatch) [inline Kudu.Services/Filters/RequestThrottleActionFilterAttribute.cs]: In the future, we will need to inspect how many sites fall into this.   I don't know if 429  from IISLog is obvious enough.   We might consider ETW tracing `once` when we get it here and reset that state when it is back to normal.

```csharp
TraceServices.CurrentRequestTracer?.Trace(...);
```

**Comment by davidebbo** [comment]
Let's not do this since ARM level throttling feels like the better solution.

**Pull request closed without merging**